ServerCore/Pages/Submissions/Index.cshtml.cs

```diff
@@ -233,38 +284,70 @@ private async Task SetupContext(int puzzleId)
-                                     {
-                                         Submission = submission,
-                                         Response = response,
-                                         SubmitterName = user.Name,
-                                         FreeformReponse = submission.FreeformResponse,
-                                         IsFreeform = Puzzle.IsFreeform
-                                     }).ToListAsync();
-
-            Submissions = new List<Submission>(SubmissionViews.Count);
+            IsPuzzleForSinglePlayer = Puzzle.IsForSinglePlayer;
+            List<SubmissionView> submissionViews = new List<SubmissionView>();
+            if (Puzzle.IsForSinglePlayer)
+            {
+                PuzzleState = await SinglePlayerPuzzleStateHelper.GetOrAddStateIfNotThere(_context,
+                    Event,
+                    Puzzle,
+                    LoggedInUser.ID);
+
+                SubmissionViews = await (from submission in _context.SinglePlayerPuzzleSubmissions
+                                         where submission.Puzzle == Puzzle
+                                            && submission.Submitter.ID == LoggedInUser.ID
+                                         join r in _context.Responses on submission.Response equals r into responses
+                                         from response in responses.DefaultIfEmpty()
+                                         orderby submission.TimeSubmitted
+                                         select new SubmissionView()
+                                         {
+                                             Submission = submission,
+                                             Response = response,
+                                             SubmitterName = LoggedInUser.Name,
+                                             FreeformReponse = submission.FreeformResponse,
+                                             IsFreeform = Puzzle.IsFreeform
+                                         }).ToListAsync();
+
+                PuzzlesCausingGlobalLockout = await SinglePlayerPuzzleStateHelper.PuzzlesCausingGlobalLockout(_context, Event, LoggedInUser.ID).ToListAsync();
+            }
+            else
+            {
+                Team = await UserEventHelper.GetTeamForPlayer(_context, Event, LoggedInUser);
+
+                PuzzleState = await (PuzzleStateHelper
+                    .GetFullReadOnlyQuery(
+                        _context,
+                        Event,
+                        Puzzle,
+                        Team))
+                    .FirstAsync();
+
+                SubmissionViews = await (from submission in _context.Submissions
+                                         where submission.Team == Team
+                                            && submission.Puzzle == Puzzle
+                                            && submission.Submitter.ID == LoggedInUser.ID
```

Comment:
I think you went too far when changing the single-player query - I should be able to see all submissions from my teammates, so the team query was good as-is.

Reply:
Fixed in newest commit